### PR TITLE
[WIP] Testing restyled kubectl ref docs

### PR DESCRIPTION
**Pull request by kbhawkey**

### CODE REVIEW

**Comment by jimangel** [comment]
Is there an example output for this anywhere @kbhawkey?

File: Makefile [hunk 1]
```diff
@@ -15,28 +15,17 @@ CLIDSTFONT=$(CLIDST)/node_modules/font-awesome
...
+
 # Build kubectl docs
 cleancli:
-	sudo rm -f main
 	sudo rm -rf $(shell pwd)/gen-kubectldocs/generators/includes
-	sudo rm -rf $(shell pwd)/gen-kubectldocs/generators/build
-	sudo rm -rf $(shell pwd)/gen-kubectldocs/generators/manifest.json
 
 cli: cleancli
+	# go mod init github.com/generators
```
Comment (tengqm) [inline Makefile]: I wonder if the output is something like what we now get from line 43.

**Comment by kbhawkey** [comment]
> Is there an example output for this anywhere @kbhawkey?

Hi @jimangel, @tengqm  . Yes, I created a PR in k8s/website with the generated output.
See https://github.com/kubernetes/website/pull/15742.
A couple of ideas I have been thinking about:
- The code needs further clean up. There is more code that can be removed/updated. I made some initial edits.
- Could the reference docs use go modules?
- Could the gen script for kubectl reside with the kubectl code, but use the hugo k8s/website reference templating to generate docs? Or leave the gen script here. Reach out to sig/cli.
- Motivation: Reduce the amount of assets that are required for the reference docs (kubectl reference, specifically) and reuse the templating and assets from k8s/website docs html generation (hugo). Create a reference doc for the kubectl commands that looks like it fits in with the rest of the k8s/website. Make it easy to generate reference docs and easy to update the command text if needed.
- The `make cli` command in this Makefile generates individual markdown files per main Commands; the sub-commands are included with the main Commands. Right now, there is a top nav list on the page to link up the sub commands. This could change if another content page template is adopted for the reference pages. I reused the reference template that I created some time ago -- could create a separate one for kubectl. I just made a few of the sections optional instead of required. There are a number of other open source reference/api docs that use a right hand side nav list.
- The flags table could use a different set of styling than the other k8s/website tables -- there are a lot of possibilities to improve the readability and navigation of the pages. The font is too small.

File: Makefile (hunk 1 above)
Comment (kbhawkey) [reply Makefile]: Yes, but with different `front matter`, heading changes, and integrated sub-commands.

**Comment by kbhawkey** [comment]
Hi, @pwittrock. This PR represents some initial ideas about aligning the generation of the `kubectl` reference doc with the kubenetes.io publishing toolchain, `hugo`. I think the `kubectl` reference doc is useful. Would you have time to provide feedback? Thank you!
Page preview:
https://deploy-preview-15742--kubernetes-io-master-staging.netlify.com/docs/reference/kubectl/kubectl-reference/alpha/

**Comment by tengqm** [comment]
@kbhawkey If you feel good about the status of the PR, we can kick it in. Approval pending from my side because there is a `[wip]` in the PR title. :)

**Comment by kbhawkey** [comment]
> 
> 
> @kbhawkey If you feel good about the status of the PR, we can kick it in. Approval pending from my side because there is a `[wip]` in the PR title. :)

hello @tengqm . I am just getting back to this. I'd like another day or so as I am investigating an `aside` toc for the pages.

File: Makefile [hunk 2]
```diff
@@ -1,42 +1,26 @@
-WEBROOT=~/src/github.com/kubernetes/website
-K8SROOT=~/k8s/src/k8s.io/kubernetes
+WEBROOT=~/tmp_kubectl_docs/src/github.com/website
+K8SROOT=~/tmp_kubectl_docs/src/k8s.io/kubernetes
```
Comment (tengqm) [inline Makefile]: you may want to reset the above two lines before checking in this file.

File: gen-kubectldocs/generators/templates.go [hunk 3]
```diff
@@ -20,49 +20,91 @@ var CategoryTemplate = `
...
 {{end}}
 
+
+<hr>
+
+
+### Version
+<div class="kubectl-reference-copyright">
+
+<a href="https://github.com/kubernetes/kubernetes">Kubectl Reference Docs version 1.15 &#xa9;Copyright 2019 The Kubernetes Authors</a>
```
Comment (tengqm) [inline gen-kubectldocs/generators/templates.go]: possible to determine the version number automatically?

File: gen-kubectldocs/generators/templates.go [hunk 4]
```diff
@@ -102,9 +102,12 @@ var CommandTemplate = `{{"---\ntitle: "}}{{.MainCommand.Name}}
 
 
 ### Version
+
 <div class="kubectl-reference-copyright">
 
-<a href="https://github.com/kubernetes/kubernetes">Kubectl Reference Docs version 1.15 &#xa9;Copyright 2019 The Kubernetes Authors</a>
+{{"<a href=\"https://github.com/kubernetes/kubernetes\">Kubectl Reference Docs  "}}
+{{"{{< param \"fullversion\" >}}  "}} {{"&#xa9;Copyright 2019 The Kubernetes Authors</a>"}}
+
```
Comment (tengqm) [inline gen-kubectldocs/generators/templates.go]: 👍

**Comment by nikhita** [comment]
:wave: just wanted to check if there are any updates here since this is blocking the migration from incubator to k-sigs :)

cc @mrbobbytables

**Comment by mrbobbytables** [comment]
pinging @jimangel to loop back in as well 👍

**Comment by kbhawkey** [comment]
> 
> 
> 👋 just wanted to check if there are any updates here since this is blocking the migration from incubator to k-sigs :)
> 
> cc @mrbobbytables

hello @nikhita . I am still (temporarily took a break) working on this. Why is the PR blocking migration? I can close and resume once the repository is moved. Where can I find information about the repo move?
Thanks!

**Comment by jimangel** [comment]
@nikhita @mrbobbytables considering the 2 open PRs are major changes with no short term plan on merging, I think we can copy the repo ASAP (let them co-exist for a finite amount of time - 2 weeks? sooner?) and set a date to close this repo.

@kbhawkey info about the move: https://github.com/kubernetes/org/issues/1158

/cc @tengqm

**Comment by kbhawkey** [comment]
[ I think we can copy the repo ASAP (let them co-exist for a finite amount of time - 2 weeks? sooner?) and set a date to close this repo.]
@jimangel , that is fine -- I still have the branch. I guess this will not wipe out my fork of kubernetes-incubator/reference-docs.

**Comment by tengqm** [comment]
@kbhawkey Maybe we can just close this PR and reopen it after the migration?

**Comment by mrbobbytables** [comment]
> @kbhawkey Maybe we can just close this PR and reopen it after the migration?

The PR should actually move over as long as the repo is migrated. You'll need to change your git config locally and set upstream to be k-sigs based instead of incubator though.

**Comment by kbhawkey** [comment]
Closing -- moving to a new location!
/close

**Comment by kbhawkey** [comment]
I plan to review and clean up the changes. If a new k8s/website theme arrives soon, I will update the related PR and output.

**Comment by kbhawkey** [comment]
Seems like this reference could be updated to use HtmlWriter and avoid generating markdown (pare down the UI and cleanup).
Though, the UI would remain unique compared to the main kubectl page (which refers to the reference sections). Not sure which way is easier to maintain.
Closing this PR (again).

**Pull request closed without merging**